### PR TITLE
feat: add snake case type keys to camel case conversion

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Compose a new type from another type by mapping all its keys from snake case to camel case

Example:
```typescript
type T = { key_of_string: string };

type U = KeysToCamelCase<T>; // Inferred as { keyOfString: string }
```

## Linear Issue Reference

https://linear.app/silverhand/issue/LOG-1427/add-snake-case-to-camel-case-type-keys-conversion

## Testing

- [x] Tested locally and it works as expected.

### CODE REVIEW

File: src/utilities/types.ts [hunk 1]
```diff
@@ -60,3 +60,20 @@ type NormalizeKeyPathsRecursively<T, K = keyof T> = K extends keyof T
...
+ */
+type CamelCase<S extends string> = S extends `${infer P1}_${infer P2}${infer P3}`
+  ? `${Lowercase<P1>}${Uppercase<P2>}${CamelCase<P3>}`
+  : Lowercase<S>;
+
+export type KeysToCamelCase<T> = {
+  [K in keyof T as CamelCase<string & K>]: T[K] extends Record<string, unknown>
+    ? KeysToCamelCase<T[K]>
+    : T[K];
+};
```
Comment (gao-sun) [inline src/utilities/types.ts:79]: looks good. any change we can test types?

**Review by gao-sun — APPROVED** [review]
lgtm

File: src/utilities/types.ts (hunk 1 above)
Comment (demonzoo) [reply src/utilities/types.ts:79]: I'm currently working on a js core feature: getTokenByRefreshToken
We will see the actual usage in that PR by EOD today.

Comment (demonzoo) [reply src/utilities/types.ts:79]: Unit test cases will be added shortly in a follow-up PR.
[Linear issue link](https://linear.app/silverhand/issue/LOG-1428/add-unit-tests-for-log-1427)

**Pull request merged**